java/src/main/java/ai/rapids/kvikio/cufile/CuFileReadHandle.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.rapids.kvikio.cufile;
+
+/**
+ * The {@code CuFileReadHandle} class represents a handle for reading data from
+ * a file using the CuFile library.
+ * <p>
+ * This class is initialized with the path to the file to be read, and it
+ * creates a native resource pointer for
+ * interacting with the file. The {@code read()} method allows reading data from
+ * the file at a specified location and
+ * offset, transferring it to a device buffer.
+ * </p>
+ */
+public final class CuFileReadHandle extends CuFileHandle {
```

Comment:
Consider merging `CuFileReadHandle` and `CuFileWriteHandle` into a single class. Since it is possible to open a file in read+write mode, I am not sure having two distinct classes is that useful. E.g., we would need a `CuFileWriteHandle.read()` and a `CuFileReadHandle.write()`?

Reply:
It is possible, however in Java it appears the canonical way to manage files is via separate read+write handles, as opposed to a shared reader+writer. There does exist this `RandomAccessFile` class in Java that supports reads+writes: https://docs.oracle.com/en/java/javase/18/docs/api/java.base/java/io/RandomAccessFile.html

My personal opinion is it would be best to support 3 versions, Read, Write, and an equivalent to `RandomAccessFile` since that appears to be more in line with Java best practices.

Reply:
Fair enough :)